okhttp/src/commonMain/kotlin/okhttp3/-CommonPlatform.kt

```diff
@@ -0,0 +1,21 @@
+/*
+ * Copyright (C) 2021 Square, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package okhttp3
+
+expect abstract class Charset
+
+expect val Charset.name: String
```

Comment:
@swankjesse Should we uplift this okio? Or is there a reason that okio doesn't consider Charset outside of jvmMain?

It's pretty central given charset is often negotiated in the response, so it needs to be a first class concept.

Reply:
I think we skip Charset completely and make multiplatform OkHttp UTF-8 only.

Reply:
Internally we need it, we might get a response with non UTF-8 encoding, and need to provide it as bytes with a Charset identifier.  So even if we avoid conversion APIs, I don't think we can avoid this completely.

Or close this and stick with no Charset in common?

Reply:
Close this, no Charset in common. As far as I know there's no generic charset encoder/decoder available in JS or native anyway, so we can't do ResponseBody.string() properly unless it's either ASCII or UTF-8. Or if we ship a dependency on ICU which I'd prefer to not.